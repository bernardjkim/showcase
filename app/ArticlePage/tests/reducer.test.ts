// import {
//   createComment,
//   createCommentError,
//   createCommentSuccess,
//   likeArticle,
//   likeArticleError,
//   likeArticleSuccess,
//   loadArticle,
//   loadArticleError,
//   loadArticleSuccess,
//   loadComments,
//   loadCommentsError,
//   loadCommentsSuccess,
//   loadLikes,
//   loadLikesError,
//   loadLikesSuccess,
// } from '../actions';
// import articlePageReducer, { initialState } from '../reducer';
// import { ArticleState } from '../types';

// describe('articlePageReducer', () => {
//   let state: ArticleState;
//   beforeEach(() => {
//     state = initialState;
//   });

//   const error = new Error('Test Error');
//   const article = {
//     _id: '_id',
//     title: 'title',
//     uri: 'uri',
//     github: 'github',
//     description: 'description',
//     tags: [],
//     image: 'image',
//   };

//   it('handles the createComment action', () => {
//     expect(articlePageReducer(state, createComment({}))).toMatchSnapshot();
//   });

//   it('handles the createCommentSuccess action', () => {
//     expect(articlePageReducer(state, createCommentSuccess({}))).toMatchSnapshot();
//   });

//   it('handles the createCommentError action', () => {
//     expect(articlePageReducer(state, createCommentError(error))).toMatchSnapshot();
//   });

//   it('handles the likeArticle action', () => {
//     expect(articlePageReducer(state, likeArticle())).toMatchSnapshot();
//   });

//   it('handles the likeArticleSuccess action', () => {
//     expect(articlePageReducer(state, likeArticleSuccess({}))).toMatchSnapshot();
//   });

//   it('handles the likeArticleError action', () => {
//     expect(articlePageReducer(state, likeArticleError(error))).toMatchSnapshot();
//   });

//   it('handles the loadArticle action', () => {
//     const query = { id: 'id' };
//     expect(articlePageReducer(state, loadArticle(query))).toMatchSnapshot();
//   });

//   it('handles the loadArticleSuccess action', () => {
//     expect(articlePageReducer(state, loadArticleSuccess(article))).toMatchSnapshot();
//   });

//   it('handles the loadArticleError action', () => {
//     expect(articlePageReducer(state, loadArticleError(error))).toMatchSnapshot();
//   });

//   it('handles the loadComments action', () => {
//     expect(articlePageReducer(state, loadComments())).toMatchSnapshot();
//   });

//   it('handles the loadCommentsSuccess action', () => {
//     expect(articlePageReducer(state, loadCommentsSuccess([]))).toMatchSnapshot();
//   });

//   it('handles the loadCommentsError action', () => {
//     expect(articlePageReducer(state, loadCommentsError(error))).toMatchSnapshot();
//   });

//   it('handles the loadLikes action', () => {
//     expect(articlePageReducer(state, loadLikes())).toMatchSnapshot();
//   });

//   it('handles the loadLikesSuccess action', () => {
//     expect(articlePageReducer(state, loadLikesSuccess([]))).toMatchSnapshot();
//   });

//   it('handles the loadLikesError action', () => {
//     expect(articlePageReducer(state, loadLikesError(error))).toMatchSnapshot();
//   });
// });
