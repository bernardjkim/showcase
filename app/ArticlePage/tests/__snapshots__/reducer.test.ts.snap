// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articlePageReducer handles the createComment action 1`] = `
Object {
  "article": undefined,
  "comments": Array [],
  "error": undefined,
  "likes": Array [],
  "loading": true,
}
`;

exports[`articlePageReducer handles the createCommentError action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": [Error: Test Error],
  "likes": Array [],
  "loading": false,
}
`;

exports[`articlePageReducer handles the createCommentSuccess action 1`] = `
Object {
  "article": undefined,
  "comments": 1,
  "error": undefined,
  "likes": Array [],
  "loading": false,
}
`;

exports[`articlePageReducer handles the likeArticle action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [],
  "loading": true,
}
`;

exports[`articlePageReducer handles the likeArticleError action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": [Error: Test Error],
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the likeArticleSuccess action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": 1,
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadArticle action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [
    Object {},
  ],
  "loading": true,
}
`;

exports[`articlePageReducer handles the loadArticleError action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": [Error: Test Error],
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadArticleSuccess action 1`] = `
Object {
  "article": Object {
    "_id": "_id",
    "description": "description",
    "github": "github",
    "image": "image",
    "tags": Array [],
    "title": "title",
    "uri": "uri",
  },
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadComments action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [
    Object {},
  ],
  "loading": true,
}
`;

exports[`articlePageReducer handles the loadCommentsError action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": [Error: Test Error],
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadCommentsSuccess action 1`] = `
Object {
  "article": undefined,
  "comments": Array [],
  "error": undefined,
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadLikes action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [
    Object {},
  ],
  "loading": true,
}
`;

exports[`articlePageReducer handles the loadLikesError action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": [Error: Test Error],
  "likes": Array [
    Object {},
  ],
  "loading": false,
}
`;

exports[`articlePageReducer handles the loadLikesSuccess action 1`] = `
Object {
  "article": undefined,
  "comments": Array [
    Object {},
  ],
  "error": undefined,
  "likes": Array [],
  "loading": false,
}
`;
